### PR TITLE
CFY-4344 Add documentation for the Psutil plugin

### PR DESCRIPTION
**Pull request by pawelcodilime**

### CODE REVIEW

File: content/plugins/psutil.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+abstract: "Cloudify Psutil plugin description and configuration"
+weight: 1150
+
+---
+{{% gsSummary %}} {{% /gsSummary %}}
+
+
+# Description
+
+The Psutil plugin can be use to collect the metrics from Windows hosts in the similar way the Diamond plugin does it on Linux hosts. It uses [psutil](https://github.com/giampaolo/psutil) underneath and can be configured to run arbitrary psutil functions. The source code is [here](https://github.com/cloudify-cosmo/cloudify-psutil-plugin).
```
Comment (abramso) [inline content/plugins/psutil.md]: The Psutil plugin can be use to collect the metrics from Windows hosts in a similar way Diamond plugin does it on Linux hosts. It uses [psutil](https://github.com/giampaolo/psutil) underneath and can be configured to run arbitrary psutil functions. The source code is [here](https://github.com/cloudify-cosmo/cloudify-psutil-plugin).

File: content/plugins/psutil.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+abstract: "Cloudify Psutil plugin description and configuration"
+weight: 1150
+
+---
+{{% gsSummary %}} {{% /gsSummary %}}
+
+
+# Description
+
+The Psutil plugin can be used to collect metrics from Windows hosts in a similar way Diamond plugin does it on Linux hosts. It uses [psutil](https://github.com/giampaolo/psutil) underneath and can be configured to run arbitrary psutil functions. The source code is [here](https://github.com/cloudify-cosmo/cloudify-psutil-plugin).
```
Comment (nir0s) [inline content/plugins/psutil.md]: "... the Diamond plugin does on Linux hosts."

File: content/plugins/psutil.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+# Usage
+
+The only necessary configuration has to be done in `node_templates`. Each node, which has to be monitored, should map:
+
+- `cloudify.interfaces.monitoring.start` to `psutil.psutil_agent.tasks.start` and
+- `cloudify.interfaces.monitoring.stop` to `psutil.psutil_agent.tasks.stop`.
+
+The first operation (`start`) accepts an input called `psutil_config` which contains the declarations of which psutil methods should be invoked and how. This input should be a list of dictionaries. Those dictionaries should have following keys:
+
+- `method` - the name of the psutil method to invoke,
```
Comment (nir0s) [inline content/plugins/psutil.md]: periods instead of commas at the end of these. thanks

File: content/plugins/psutil.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                args:
+                  interval: 1
+                  percpu: false
+                interval: 5
+                result_argument: user
+                alias: cpu_total_user
+        stop: psutil.psutil_agent.tasks.stop
+{{< /gsHighlight >}}
+
+In this example for the node `vm` two metrics will be collected:
```
Comment (nir0s) [inline content/plugins/psutil.md]: In this example, two metrics will be collected for the node `vm`:

File: content/plugins/psutil.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+- free virtual memory (psutil method `virtual_memory` and result argument `free`) each 5 seconds and will be sent as `memory_MemFree`,
```
Comment (nir0s) [inline content/plugins/psutil.md]: each -> every

File: content/plugins/psutil.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+- user CPU usage in percent (psutil method `cpu_times_percent` and result argument `user` invoked with arguments `interval=1` and `percpu=False`) each 5 seconds and will be sent as `cpu_total_user`.
```
Comment (nir0s) [inline content/plugins/psutil.md]: eac h-> every

File: content/plugins/psutil.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+Keep in mind that each parameter will be converted to corresponding Python type (in particular `false` in YAML will be converted to `False` in Python).
```
Comment (nir0s) [inline content/plugins/psutil.md]: the corresponding

File: content/plugins/psutil.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+- `loop.py` - it contains one method, `collect_metrics()` which handles actual metric collection.
+
+## tasks.py
+
+### start()
+
+First, this method retrieves all of the necessary data from the context (like deployment ID or broker credentials). Then it takes the configuration from the blueprint and serializes it all to JSON. After that it creates a service using NSSM with those parameters. Lastly, it starts the service.
+
+### stop()
+This method stops the service and then deletes it.
```
Comment (nir0s) [inline content/plugins/psutil.md]: just wondering.. why is there no "Create" and "Delete"?

**Pull request closed without merging**